[Bug] proxy DNS UDP timed out after 5000  milliseconds  error in kubernetes

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.


### Version

pulsar 3.3.0
kubernetes 1.18.4 


### Minimal reproduce step

online error, reproduction is not easy


### What did you expect to see?

consume success

### What did you see instead?

``` java
java.util.concurrent.CompletionException: io.netty.resolver.dns.DnsResolveContext$SearchDomainUnknownHostException: Failed to resolve 'pulsar-uqhf-178-pulsar-cluster-broker-1.pulsar-uqhf-178-pulsar-cluster-broker-headless.pulsar.svc.cluster.local' [A(1)] and search domain query for configured domains failed as well: [pulsar.svc.cluster.local, svc.cluster.local, cluster.local]
Caused by: io.netty.resolver.dns.DnsResolveContext$SearchDomainUnknownHostException: Failed to resolve 'pulsar-uqhf-178-pulsar-cluster-broker-1.pulsar-uqhf-178-pulsar-cluster-broker-headless.pulsar.svc.cluster.local' [A(1)] and search domain query for configured domains failed as well: [pulsar.svc.cluster.local, svc.cluster.local, cluster.local]
Caused by: io.netty.resolver.dns.DnsNameResolverTimeoutException: [37005: /9.165.255.254:53] DefaultDnsQuestion(pulsar-uqhf-178-pulsar-cluster-broker-1.pulsar-uqhf-178-pulsar-cluster-broker-headless.pulsar.svc.cluster.local. IN A) query '37005' via UDP timed out after 5000 milliseconds (no stack trace available)
2024-10-20T15:18:34,102+0800 [pulsar-proxy-io-2-5] INFO  org.apache.pulsar.proxy.server.ProxyConnection - [/21.10.52.200:38756] Connection closed

```
this error happen in proxy and then some consumers will disconnect and close the client

### Anything else?

With the increase in traffic, we have recently discovered that this problem will occasionally occur in the cluster. And there are many consumers using reader mode in our cluster.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java
+++ b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java
@@ -188,7 +188,9 @@ public class ConnectionPool implements AutoCloseable {
     private static AddressResolver<InetSocketAddress> createAddressResolver(ClientConfigurationData conf,
                                                                             EventLoopGroup eventLoopGroup) {
         DnsNameResolverBuilder dnsNameResolverBuilder = new DnsNameResolverBuilder()
-                .traceEnabled(true).channelType(EventLoopUtil.getDatagramChannelClass(eventLoopGroup));
+                .traceEnabled(true)
+                .channelType(EventLoopUtil.getDatagramChannelClass(eventLoopGroup))
+                .socketChannelType(EventLoopUtil.getClientSocketChannelClass(eventLoopGroup), true);
         if (conf.getDnsLookupBindAddress() != null) {
             InetSocketAddress addr = new InetSocketAddress(conf.getDnsLookupBindAddress(),
                     conf.getDnsLookupBindPort());

--- a/pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyService.java
+++ b/pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyService.java
@@ -179,7 +179,8 @@ public class ProxyService implements Closeable {
         this.authenticationService = authenticationService;
 
         DnsNameResolverBuilder dnsNameResolverBuilder = new DnsNameResolverBuilder()
-                .channelType(EventLoopUtil.getDatagramChannelClass(workerGroup));
+                .channelType(EventLoopUtil.getDatagramChannelClass(workerGroup))
+                .socketChannelType(EventLoopUtil.getClientSocketChannelClass(workerGroup), true);
         DnsResolverUtil.applyJdkDnsCacheSettings(dnsNameResolverBuilder);
 
         dnsAddressResolverGroup = new DnsAddressResolverGroup(dnsNameResolverBuilder);

